### DISCUSSION TITLE
Non abstract empty bodyed constructors?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [dotnet/csharplang#8228] [Language Ideas]
Can we get bodyless constructors for non abstract classes for situations where we just need to pass something to base()?

```cs
public class Createable<T>(Func<T> Create)
{
     protected T Value => Create();
}

public class HashCode : Createable<int>
{
   public HashCode(string hashCode) : base(() =>
   {
       return hasCode.GetHashCode()
   });
}
```

it would simply act like this...
```cs
public class HashCode : Createable<int>
{
   public HashCode(string hashCode) : base(() =>
   {
       return hasCode.GetHashCode()
   }) { }
}
```

Note these are just example classes, I am actually using this pattern for Textures

**Comment by KennethHoff** [comment]
Primary constructors? 

```cs
public class HashCode(string hashCode) : Createable<int>(() =>
   {
       return hasCode.GetHashCode()
   });
```
Granted, you now could theoretically have access to that hashCode for the lifetime of the object ( If you access it, that is), so it's not ideal in that sense, but it's much less code than even your suggestion.

---
I agree with the general idea though, I think everywhere you have to add some random character(s) for seemingly no reason they should be optional.

**Reply by Shadowblitz16** [reply]
Well this works unless I need to define methods or alternet constructors then all of a sudden it's this again..
```cs
public class HashCode(string hashCode) : Createable<int>(() =>
   {
       return hasCode.GetHashCode()
   })
{

}
```
